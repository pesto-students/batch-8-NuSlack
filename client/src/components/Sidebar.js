import React, { Fragment } from "react";
import styled from "styled-components";
import Channels from "./Channels";
import Messages from "./Messages";

const Container = styled.div`
  height: 100%;
  background: rebeccapurple;
  padding: 1rem;
  color: white;
  box-sizing: border-box;
`;
const SidebarHeader = styled.header`
  display: grid;
  grid-template-columns: 1fr 25px;
  font-size: 1.2rem;
`;

const MainHeading = styled.h1`
  font-weight: 900;
  font-size: 1.3rem;
`;
const MemberNameContainer = styled.div`
  font-size: 1rem;
  grid-column-start: 1;
  grid-column-end: 3;
  margin-top: 0.5rem;
`;
export const Status = styled.span`
  height: 0.7rem;
  width: 0.7rem;
  border-radius: 100%;
  background-color: green;
  margin-right: 0.5rem;
  display: inline-block;
`;
function Sidebar() {
  return (
    <Container>
      <SidebarHeader>
        <MainHeading>Lorem Ipsum</MainHeading>
        <div>
          <i className="far fa-bell" />
        </div>
        <MemberNameContainer>
          <Status />
          Member 1
        </MemberNameContainer>
      </SidebarHeader>
      <Fragment>
        <Channels />
        <Messages />
      </Fragment>
    </Container>
  );
}

export default Sidebar;
