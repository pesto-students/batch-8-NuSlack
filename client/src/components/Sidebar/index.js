import React, { useState, useEffect } from 'react';
import { Layout, Menu, Icon } from 'antd';
import axios from 'axios';
import { useHomeContext } from '../../context/HomeContext';
import AddChannelModal from '../AddChannelModal'
const { SubMenu } = Menu;
const { Sider } = Layout;

const Sidebar = () => {
  const [userChannels, setUserChannels] = useState([]);
  const [groupChannels, setGroupChannels] = useState([]);
  const [channelModalIsVisible, setChannelModalIsVisible] = useState(false);

  const { user } = useHomeContext();
  useEffect(() => {
    setGroupChannels([
      { name: 'channel1', id: '123123' },
      { name: 'channel2', id: '123122' },
    ]);
    setUserChannels([
      { name: 'Person1', id: '1223' },
      { name: 'Person2', id: '1122' },
    ]);
    axios
      .get('http://localhost:8080/channels', {
        params: {
          users: user._id,
        },
      })
      .then(res => {
        const { data: channels } = res;
        console.log(channels);
        const userChannels = channels.filter(
          channel => channel.isGroup === false,
        );
        const groupChannels = channels.filter(
          channel => channel.isGroup === true,
        );
        setGroupChannels(groupChannels);
        setUserChannels(userChannels);
      });
  }, [user]);
  const toggleModal = () => {
    console.log('asas')
    setChannelModalIsVisible(!channelModalIsVisible)
  }
  return (
    <Sider
      width={300}
      style={{
        overflow: 'hidden',
        height: '100%',
      }}
      breakpoint="sm"
      collapsible
      theme="light"
    >
      <Menu
        mode="inline"
        defaultSelectedKeys={['1']}
        defaultOpenKeys={['sub1']}
        style={{ height: '100%', borderRight: 0 }}
      >
        <SubMenu
          key="sub1"
          title={
            <span>
              <Icon type="laptop" />
              Channels <Icon type="plus" onClick={toggleModal}/>
            </span>
          }
        >
          {groupChannels.map(channel => (
            <Menu.Item key={channel.id}>{channel.name}</Menu.Item>
          ))}
        </SubMenu>
        <SubMenu
          key="sub2"
          title={
            <span>
              <Icon type="user" />
              People
            </span>
          }
        >
          {userChannels.map(channel => (
            <Menu.Item key={channel.id}>{channel.name}</Menu.Item>
          ))}
        </SubMenu>
      </Menu>
      <AddChannelModal visible={channelModalIsVisible} toggleModal={toggleModal}/>
    </Sider>
  );
};

export default Sidebar;
