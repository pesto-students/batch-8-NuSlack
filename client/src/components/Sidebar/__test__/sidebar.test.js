import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Sidebar from '../index';

describe('<Sidebar />', () => {
  it('should match the snapshot', () => {
    // FIXME
    const tree = shallow(<Sidebar size="default" />);
    expect(toJson(tree)).toMatchSnapshot();
  });
});
