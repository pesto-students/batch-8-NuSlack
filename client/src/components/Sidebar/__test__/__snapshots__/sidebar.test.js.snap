// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Sidebar /> should match the snapshot 1`] = `
<SideBarContainer>
  <SubMenu
    key="sub1"
    title={
      <span>
        <Icon
          type="usergroup-add"
        />
        <span>
          Channels 
          <Icon
            onClick={[Function]}
            type="plus"
          />
        </span>
      </span>
    }
  />
  <SubMenu
    key="sub2"
    title={
      <span>
        <Icon
          type="user"
        />
        <span>
          People
        </span>
      </span>
    }
  />
  <Complete />
  <AddChannelModal
    toggleModal={[Function]}
    visible={false}
  />
</SideBarContainer>
`;
